include/nbl/builtin/hlsl/workgroup2/shared_scan.hlsl

```diff
@@ -43,10 +43,8 @@ struct reduce<Config, BinOp, 1, device_capabilities>
 
         subgroup2::reduction<params_t> reduction;
         vector_t value;
-        dataAccessor.template get<vector_t>(workgroup::SubgroupContiguousIndex(), value);
-        value = reduction(value);
-        return value[0];
-        // dataAccessor.template set<vector_t>(workgroup::SubgroupContiguousIndex(), value);
+        dataAccessor.template get<vector_t, uint32_t>(glsl::gl_SubgroupInvocationID(), value);
```

Comment:
I think you can safely use `uint16_t` for the index for anything using an accessor in this file

Reply:
`glsl::gl_SubgroupInvocationID()` returns a `uint16_t` though.

Reply:
yep use `uint16_t` throughout for indexing, realistically even if you end up with a 16k virtual workgroup you're safe because max index is 64k-1